h5py/_hl/group.py

```diff
@@ -499,7 +499,10 @@ def copy(self, source, dest, name=None,
                     dest_path = name
                 else:
                     # copy source into dest group: dest_name/source_name
-                    dest_path = pp.basename(h5i.get_name(source[source_path].id))
+                    if isinstance(source, dataset.Dataset):
```

Comment:
I think we should actually check `source == '.'` here, rather than its type. The same problem would also affected named datatypes stored in the file.

Reply:
Also, we can use an `elif` to have one less layer of nesting.